langkit/compile_context.py

```diff
@@ -1702,23 +1717,36 @@ def emit(self, lib_root, check_only=False, warnings=None,
         # Compute the list of passes to run:
 
         # First compile the DSL
-        all_passes = self.compilation_passes
+        self.all_passes = self.compilation_passes
 
         # Then, if requested, emit code for the generated library
-        if not check_only:
-            all_passes.append(
+        if not self.check_only:
+            self.all_passes.append(
                 self.prepare_code_emission_pass(lib_root, **kwargs))
 
-            all_passes.extend(self.code_emission_passes(annotate_fields_types))
+            self.all_passes.extend(self.code_emission_passes())
 
             # Run plugin passes at the end of the pipeline
-            all_passes.extend(plugin_passes)
+            self.all_passes.extend(plugin_passes)
+
+        for p in self.all_passes:
+            if p.is_optional and p.name in explicit_passes_triggers.keys():
+                trig = explicit_passes_triggers[p.name]
+                printcol(
```

Comment:
Could we enable this printing only when verbosity level is “debug”?

Reply:
I don't know, if you don't add that you can pass anything to `--pass-on/off` and you won't know if it's working or not. Another solution would be to emit an error message if the user passes a wrong pass name, but I prefered the info message solution

Reply:
I think we should definitely stop with an error if there is a typo in the name of the pass (just like we must stop with an error if there is a typo in an option name rather than ignore it with a warning). But then indeed no reason to print the enabling status for all optional passes.

Reply:
OK, I'll do that